crates/eips/src/eip2930.rs

```diff
@@ -141,6 +141,31 @@ pub struct AccessListWithGasUsed {
     pub gas_used: U256,
 }
 
+
+/// `AccessListResult` for handling errors from `eth_createAccessList`
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct AccessListResult {
+    /// List with accounts accessed during transaction.
+    pub access_list: AccessList,
+    /// Estimated gas used with access list.
+    pub gas_used: U256,
+    /// Optional error message if the transaction failed.
+    #[cfg_attr(feature = "serde", serde(default, skip_serializing_if = "Option::is_none"))]
+    pub error: Option<String>,
+}
+
+impl AccessListResult {
+    pub fn ensure_ok(self) -> Result<AccessListWithGasUsed, String> {
+        unimplemented!()
```

Comment:
missing impl and docs

Reply:
https://github.com/alloy-rs/alloy/pull/1110/commits/93c58717c10196317ab1d08962e92caea2cf4309